packages/deployments/contracts/contracts/messaging/connectors/SpokeConnector.sol

```diff
@@ -255,7 +272,7 @@ abstract contract SpokeConnector is Connector, ConnectorManager, WatcherClient,
    * @notice This returns the root of all messages with the origin domain as this domain (i.e.
    * all outbound messages)
    */
-  function send() external whenNotPaused {
+  function send() external whenNotPaused rateLimited {
     bytes memory _data = abi.encodePacked(MERKLE.root());
```

Comment:
Don't we need to actually check against `sentMessageRoots`? I think this is incomplete.

Reply:
I fixed this, blz check @jakekidd

Reply:
yep, gob job, ty